### ISSUE TITLE
[Q] How to store images as files and efficiently reference them in tables?

### ISSUE THREAD

**Issue by collinmccarthy** [wandb/wandb#4097]
Hello,

I would like to save images into a directory so I have the raw images available locally (save the file locally then call `wandb.save()`), and then I would also like to log references to these images into tables (use `image = wandb.Image(filepath)` and `wandb.Table(image)`. I don't want the image data repeated / stored twice, since the images are relatively large (several MB each). I'm planning on storing a few hundred images per run.

Will this approach above avoid data being stored twice? I know I can use the API to extract the images from the table but I want to store the images as raw files for easier post-processing. 

Thanks!

**Comment by MBakirWB** [comment]
Hi @collinmccarthy , not sure if what you had in mind is the most efficient way of working with an image dataset. You can use `artifact.add() `to log tables to the Artifacts section of your run instead of the workspace. This could be useful if you have a dataset that you want to log once and then reference for future runs. See [here](https://docs.wandb.ai/guides/data-vis/log-tables#advanced-artifact-tables) in our docs, and additionally this report example on the above. Please let me know if this works for you use case.

**Comment by MBakirWB** [comment]
Hi @collinmccarthy , following up on this request. Do you still need assistance with your logging images workflow
?

**Comment by MBakirWB** [comment]
Hi @collinmccarthy, since we have not heard back from you we are going to close this request. If you would like to re-open the conversation, please let us know!

**Comment by collinmccarthy** [comment]
Hi @MBakirWB sorry for the delay here. A couple of things to clarify my goals:
- I do want to log to the workspace so I can see / compare tables across runs easily
- I am going to log this image every run (to see how good my segmentation predictions are)

An ideal workflow for me would be the following:
- Create an image and save it locally to `filepath`
- Log the image directly to my files and get the saved path with `saved_image= wandb.save(filepath)`
- Use that previous path to add the file to a table in the workspace with `wandb.Table(saved_image)`

I would hope the `wandb.Table` call would recognize that the path I gave it was within a `wandb` subfolder and therefore not create a second image (png) file in my `media/` directory. There would only be one version of the file in my files folder for the run, from when I called `wandb.save()`. Does this make sense?

Anyway, it's not the biggest deal, and from what I can tell this is more of a feature request. At a higher level I want to be able to create `wandb.Table` objects in my workspace that are simply different ways of viewing the data that's already there, not duplicating it. But I can imagine that would be a more difficult change to implement so I understand if this is not a realistic request. I'll go ahead and close this ticket since my question is answered. Thank you!
